src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -294,6 +314,27 @@ private static (RoutePattern resolvedRoutePattern, IDictionary<string, string> r
 
             builder.Metadata.Add(action);
 
+            // MVC guarantees that when two of it's endpoints have the same route name they are equivalent.
+            //
+            // The case for this looks like:
+            //
+            //  [HttpGet]
+            //  [HttpPost]
+            //  [Route("/Foo", Name = "Foo")]
+            //  public void DoStuff() { }
+            //
+            // However, Endpoint Routing requires Endpoint Names to be unique.
+            //
+            // We can use the route name as the endpoint name if it's not set. Note that there's no
+            // attribute for this today so it's unlikley. Using endpoint name on a 
+            if (routeName != null && 
+                !suppressLinkGeneration && 
+                routeNames.Add(routeName) &&
+                builder.Metadata.OfType<IEndpointNameMetadata>().FirstOrDefault()?.EndpointName == null)
```

Comment:
Because the last metadata wins, should this be LastOrDefault?

Reply:
ya good point.